openedx/core/lib/xblock_builtin/xblock_discussion/tests.py

```diff
@@ -0,0 +1,136 @@
+    """
+    Import and export tests
+    """
+    DISCUSSION_XBLOCK_LOCATION = "openedx.core.lib.xblock_builtin.xblock_discussion.xblock_discussion.DiscussionXBlock"
+
+    @staticmethod
+    def _make_xml_node(xml_string):
+        """
+        Builds etree Element from string XML
+        """
+        return etree.fromstring(xml_string)
+
+    def setUp(self):
+        """
+        Set up method
+        """
+        super(DiscussionXBlockImportExportTests, self).setUp()
+        self.keys = ScopeIds("any_user", "discussion", "def_id", "usage_id")
+        self.runtime_mock = mock.Mock(spec=Runtime)
+        self.runtime_mock.construct_xblock_from_class = mock.Mock(side_effect=self._construct_xblock_mock)
+        self.runtime_mock.get_policy = mock.Mock(return_value={})
+        self.id_gen_mock = mock.Mock()
+
+    def _construct_xblock_mock(self, cls, keys):  # pylint: disable=unused-argument
+        """
+        Builds target xblock instance (DiscussionXBlock)
+
+        Signature-compatible with runtime.construct_xblock_from_class - can be used as a mock side-effect
+        """
+        return DiscussionXBlock(self.runtime_mock, scope_ids=keys, field_data=DictFieldData({}))
+
+    @mock.patch(DISCUSSION_XBLOCK_LOCATION + ".load_definition_xml")
+    @ddt.unpack
+    @ddt.data(
+        ("ID1", "Test category", "Test target"),
+        ("ID2", "Some other category", "Some other target")
+    )
+    def test_xblock_export_format(self, discussion_id, category, target, patched_load_definition_xml):
+        """
+        Test that xblock export XML format can be parsed preserving field values
+        """
+        xblock_xml = """
+        <discussion
+            url_name="82bb87a2d22240b1adac2dfcc1e7e5e4" xblock-family="xblock.v1"
+            discussion_id="{discussion_id}"
+            discussion_category="{category}"
+            discussion_target="{target}"
+        />
+        """.format(discussion_id=discussion_id, category=category, target=target)
+        node = self._make_xml_node(xblock_xml)
+
+        patched_load_definition_xml.side_effect = Exception("Irrelevant")
+
+        block = DiscussionXBlock.parse_xml(node, self.runtime_mock, self.keys, self.id_gen_mock)
+        self.assertEqual(block.discussion_id, discussion_id)
+        self.assertEqual(block.discussion_category, category)
+        self.assertEqual(block.discussion_target, target)
+
+    @mock.patch(DISCUSSION_XBLOCK_LOCATION + ".load_definition_xml")
+    @ddt.unpack
+    @ddt.data(
+        ("ID1", "Test category", "Test target"),
+        ("ID2", "Some other category", "Some other target")
+    )
+    def test_legacy_export_format(self, discussion_id, category, target, patched_load_definition_xml):
+        """
+        Test that legacy export XML format can be parsed preserving field values
+        """
+        xblock_xml = """
+            <discussion url_name="82bb87a2d22240b1adac2dfcc1e7e5e4" discussion_id="{discussion_id}"/>
```

Comment:
`discussion_id` never appeared in the old XML, but `id` potentially could (it can also be in `policy.json`). Likewise, `for` in the xml should get mapped to `discussion_target` on the XBlock. See https://github.com/edx/edx-platform/commit/33b3dfcc9279ce59a1b8adf49adb638202d8680f#diff-6e99382019ec7061fa7a155bea7d251fL117